Cython/Compiler/ModuleNode.py

```diff
@@ -3931,7 +3931,8 @@ def generate_exttype_vtable_init_code(self, entry, code):
                 if entry.func_cname]
             if c_method_entries:
                 for meth_entry in c_method_entries:
-                    cast = meth_entry.type.signature_cast_string()
+                    vtable_type = getattr(meth_entry, 'vtable_type', meth_entry.type)
```

Comment:
What we usually do instead of a `getattr()` is to add the new attribute normally as a class (default) attribute in `Entry` and then set it only when we have something reasonable to assign it to. That way, it always exists.